### ISSUE TITLE
[Accessibility]: Upload/Index timeline images have no semantics/context

### ISSUE THREAD

**Issue by Strum355** [sourcegraph/sourcegraph#35261]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33513, #33512

### Problem description

Each of the timeline images is announced as just "image", with the state being conveyed by the image not spoken by the screen reader. 

![image](https://user-images.githubusercontent.com/18282288/167699579-4879e664-17bc-497d-9834-2729c2467d41.png)

Timelines with failed steps dont clearly announce the state of any individual step either, making it unclear which step failed

![image](https://user-images.githubusercontent.com/18282288/167701657-2af5ab26-73eb-49c8-8fd2-877a2a02a80a.png)

Somewhat related issue here: https://github.com/sourcegraph/sourcegraph/issues/35146 
cc @sourcegraph/frontend-platform 

### Expected behavior

We are undecided on whether it would be better to skip the images and update the text to announce the step state, or announce the state in the image.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

I'm unsure who should work on this issue or if it needs more clarification. I will let the Frontend Platform team triage this.

**Comment by umpox** [comment]
@gitstart-sourcegraph Please look at this as it's related to the recent `<Icon />` accessibility work. See this Storybook path for an example of this component: https://storybook.sgdev.org/?path=/story/web-codeintel-indexes-codeintelindexpage--completed
